### ISSUE TITLE
MemcachedClient.Remove fails to remove items from cache

### ISSUE THREAD

**Issue by Gonyoda** [cnblogs/EnyimMemcachedCore#64]
EnyimMemcachedCore v2.1.8

ExecuteRemoveAsync doesn't hash the key before determine the pool node.  Therefore, the remove always fails.

```
public async Task<IRemoveOperationResult> ExecuteRemoveAsync(string key)
    {
      IMemcachedNode memcachedNode = this.pool.Locate(key);
```

should be:
```
public async Task<IRemoveOperationResult> ExecuteRemoveAsync(string key)
    {
      string hashedKey = this.keyTransformer.Transform(key); 
      IMemcachedNode memcachedNode = this.pool.Locate(hashedKey );
```

And similar for the non-async version.  And, hashedKey needs to be passed to OperationFactory.Delete later in the method(s).

The key is hashed in the PerformStoreAsync and PerformTryGet.

I'd make a PR to help out, but I'm not sure which branch to work off.

**Comment by cnblogs-dudu** [comment]
From master branch.

**Comment by Gonyoda** [comment]
A work-around is to store a reference to the KeyTransformer used in the MemcachedClientConfiguration and transforming the key prior to calling Remove:

```
_keyTransformer = new TigerHashKeyTransformer();
...
configuration.KeyTransformer = _keyTransformer;
IDistributedCache _distributedCache = new MemCacheClient(..., configuration);
...
_distributedCache.RemoveAsync(_keyTransformer.Transform(key));
```

I'm not sure how to work-around using DI...

**Comment by Gonyoda** [comment]
Fixed in nuget release [2.1.19](https://www.nuget.org/packages/EnyimMemcachedCore/2.1.9)
